fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -180,9 +180,124 @@ def additional_upload_parameters
     end
   end
 
+  # Generates commands and executes the altool.
+  class AltoolTransporterExecutor < TransporterExecutor
+    ERROR_REGEX = /\sNSLocalizedFailureReason\s=\s"+(.+)"/
+
+    private_constant :ERROR_REGEX
+
+    def execute(command, hide_output)
+      if Helper.test?
+        yield(nil) if block_given?
+        command
+      end
+
+      @errors = []
+      @warnings = []
+      @all_lines = []
+
+      if hide_output
+        # Show a one time message instead
+        UI.success("Waiting for App Store Connect transporter to be finished.")
+        UI.success("Application Loader progress... this might take a few minutes...")
+      end
+
+      begin
+        exit_status = FastlaneCore::FastlanePty.spawn(command) do |command_stdout, command_stdin, pid|
+          begin
+            command_stdout.each do |line|
+              @all_lines << line
+              parse_line(line, hide_output) # this is where the parsing happens
+            end
+          end
+        end
+      rescue => ex
+        # FastlanePty adds exit_status on to StandardError so every error will have a status code
+        exit_status = ex.exit_status
+        @errors << ex.to_s
+      end
+
+      @errors << "The call to the altool completed with a non-zero exit status: #{exit_status}. This indicates a failure." unless exit_status.zero?
+
+      UI.important(@warnings.join("\n")) unless @warnings.empty?
```

Comment:
> And as a bonus you would not have to test for @warnings.empty? since the each loop would just never iterate if the array is empty 🙃

Ah, yes, currently there is no change to parse warning here, so maybe it would be better to remove this line

Reply:
https://github.com/fastlane/fastlane/pull/20631/commits/44bf7aaf775a90cf9e02101bc070167653fe2bc4 fixed!